apps/src/templates/census2017/CensusTeacherBanner.jsx

```diff
@@ -0,0 +1,350 @@
+    marginRight: 20,
+  },
+  introQuestion: {
+    marginBottom: 5
+  },
+  main: {
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: color.teal,
+    minHeight: 72,
+    width: styleConstants['content-width'],
+    backgroundColor: color.white,
+    overflowWrap: 'break-word'
+  },
+  message: {
+    marginTop: 0,
+    marginBottom: 20,
+    marginLeft: 20,
+    marginRight: 20,
+  },
+  radio: {
+    verticalAlign: 'top',
+    marginRight: 10,
+  },
+  title: {
+    marginBottom: 0,
+  },
+  updateSchool: {
+    fontSize: '85%',
+    marginTop: 0,
+    marginBottom: 0,
+  },
+};
+
+export default class CensusTeacherBanner extends Component {
+  static propTypes = {
+    schoolYear: PropTypes.number.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    onDismiss: PropTypes.func.isRequired,
+    onPostpone: PropTypes.func.isRequired,
+    onChange: PropTypes.func.isRequired,
+    ncesSchoolId: PropTypes.string.isRequired,
+    question: PropTypes.oneOf(['how_many_10_hours', 'how_many_20_hours']).isRequired,
+    selection: PropTypes.bool,
+    teacherId: PropTypes.number.isRequired,
+    teacherName: PropTypes.string.isRequired,
+    teacherEmail: PropTypes.string.isRequired,
+    showUnknownError: PropTypes.bool,
+  };
+
+  constructor() {
+    super();
+    this.bindSchoolInfoInputs = this.bindSchoolInfoInputs.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadSchoolName(this.props.ncesSchoolId);
+  }
+
+  state = {
+    showSchoolInfoForm: false,
+    country: 'United States',
+    schoolType: '',
+    ncesSchoolId: null,
+    schoolName: '',
+    schoolDisplayName: null,
+    schoolState: '',
+    schoolZip: '',
+    schoolLocation: '',
+  };
+
+  onCountryChange(_, event) {
+    const newCountry = event ? event.value : '';
+    this.setState({country: newCountry});
+  }
+
+  onSchoolTypeChange(event) {
+    const newType = event ? event.target.value : '';
+    this.setState({schoolType: newType});
+  }
+
+  onSchoolChange(_, event) {
+    const newSchool = event ? event.value : '';
+    this.setState({ncesSchoolId: newSchool});
+    this.loadSchoolName(newSchool);
+  }
+
+  onSchoolNotFoundChange(field, event) {
+    const newValue = event ? event.target.value : '';
+    this.setState({
+      [field]: newValue
+    });
+  }
+
+  showSchoolInfoForm() {
+    this.setState({showSchoolInfoForm: true});
+  }
+
+  hideSchoolInfoForm() {
+    this.setState({
+      showSchoolInfoForm: false,
+      showSchoolInfoErrors: false,
+    });
+  }
+
+  dismissSchoolInfoForm() {
+    // Clear any state that may have been set
+    this.setState({
+      showSchoolInfoForm: false,
+      country: 'United States',
+      schoolType: '',
+      ncesSchoolId: null,
+      schoolName: '',
+      schoolDisplayName: null,
+      schoolState: '',
+      schoolZip: '',
+      schoolLocation: '',
+      showSchoolInfoErrors: false,
+      showSchoolInfoUnknownError: false,
+    });
+    this.loadSchoolName(this.props.ncesSchoolId);
+  }
```

Comment:
Is this needed? Will the dismissal also unmount the component? If we unmount the component on  @dismissal, and set default state (line 84) when it's new, then do we need to clear it on the way out?

Reply:
Yes, this is needed. This is handling the case where the user dismissed the "update your school" view, not the entire banner. In that case, there may have been partial state about the school changed. We do not want to use that partial state in the census submission so we need to reset to the previous values. I'll add a comment to make this clearer.